src/network/RTypeClient.cpp

```diff
@@ -0,0 +1,166 @@
+    if ((size_t)entityId != entity) {
+        std::cerr << "Error: Entity created does not have the same id as received in the network" << std::endl;
+    }
+} 
+
+void RTypeClient::deleteEntity(Engine::GameEngine &engine, std::vector<uint8_t> operation)
+{
+    uint16_t entityId = uint16From2Uint8(operation[1], operation[2]);
+    ECS::Entity entity = engine.getRegistry().entityManager().entityFromIndex(entityId);
+
+    engine.getRegistry().entityManager().killEntity(entity);
+}
+
+void RTypeClient::attachComponent(Engine::GameEngine &engine, std::vector<uint8_t> operation)
+{
+    uint16_t entityId = uint16From2Uint8(operation[1], operation[2]);
+    uint16_t componentId = uint16From2Uint8(operation[3], operation[4]);
+    std::string &strCompId = _compNames[componentId];
+    std::type_index compTypeIndex = engine.getTypeIndexFromString(strCompId);
+    
+    auto &compInstance = engine.getComponentFromId(strCompId);
+    auto &sparseArray = compInstance->any_cast(
+        engine.getRegistry().componentManager().getComponents(compTypeIndex)
+    );
+    sparseArray.constructAt(entityId);
+}
+
+void RTypeClient::updateComponent(Engine::GameEngine &engine, std::vector<uint8_t> operation)
+{
+    uint16_t entityId = uint16From2Uint8(operation[1], operation[2]);
+    uint16_t componentId = uint16From2Uint8(operation[3], operation[4]);
+    std::string strCompId = _compNames[componentId];
+    std::type_index compTypeIndex = engine.getTypeIndexFromString(strCompId);
+    
+    auto &compInstance = engine.getComponentFromId(compTypeIndex);
+    auto &sparseArray = compInstance->any_cast(
+        engine.getRegistry().componentManager().getComponents(compTypeIndex)
+    );
+    std::vector<uint8_t> serializedData = std::vector<uint8_t>(operation.begin() + 5, operation.end());
+    sparseArray[entityId]->deserialize(serializedData);
+}
+
+void RTypeClient::detachComponent(Engine::GameEngine &engine, std::vector<uint8_t> operation)
+{
+    uint16_t entityId = uint16From2Uint8(operation[1], operation[2]);
+    uint16_t componentId = uint16From2Uint8(operation[3], operation[4]);
+    std::string strCompId = _compNames[componentId];
+    std::type_index compTypeIndex = engine.getTypeIndexFromString(strCompId);
+    
+    auto &compInstance = engine.getComponentFromId(compTypeIndex);
+    auto &sparseArray = compInstance->any_cast(
+        engine.getRegistry().componentManager().getComponents(compTypeIndex)
+    );
+    sparseArray.erase(entityId);
+}
+
+uint16_t RTypeClient::uint16From2Uint8(uint8_t first, uint8_t second)
+{
+    return static_cast<uint16_t>((first << 8) | static_cast<uint16_t>(second));
+}
+
+uint16_t RTypeClient::receiveUint16()
+{
+    std::vector<uint8_t> data = blockingReceive();
+
+    if (data.size() < 2) {
+        throw std::runtime_error("Received data is too short to make an uint16");
+    }
+    uint16_t result = (static_cast<uint16_t>(data[0]) << 8) | static_cast<uint16_t>(data[1]);
```

Comment:
Coundt you use your uint16From2Uint8 here for data[0] and data[1] ?

Reply:
Done